socs/agents/hwp_supervisor/agent.py

```diff
@@ -1143,6 +1163,42 @@ def check_acu_ok_for_spinup():
                         freq_tol_duration=30,
                     ))
                     return
+            elif isinstance(state, ControlState.FastBrake):
+                # Flip PID direciton and tune stop
+                pid_dir = int(query_pid_state()['direction'])
+                if pid_dir == 1:
+                    self.run_and_validate(
+                        clients.pcu.send_command,
+                        kwargs={'command': 'on_2'}, timeout=None
+                    )
+                else:
+                    self.run_and_validate(
+                        clients.pcu.send_command,
+                        kwargs={'command': 'on_1'}, timeout=None
+                    )
+                self.run_and_validate(clients.pmx.ign_ext)
+                self.run_and_validate(clients.pmx.set_v, kwargs={'volt': state.brake_voltage})
+                self.run_and_validate(clients.pmx.set_on)
+
+                time.sleep(10)
+                self.action.set_state(ControlState.WaitForFastBrake(
+                    min_freq=0.2,
+                    prev_freq=hwp_state.enc_freq,
+                ))
+
+            elif isinstance(state, ControlState.WaitForFastBrake):
+                f0 = query_pid_state()['current_freq']
+                time.sleep(5)
+                f1 = query_pid_state()['current_freq']
+                if f0 < 0.2 or (f1 > f0):
+                    self.log.info("Turning off PMX and putting PCU in stop mode")
+                    self.run_and_validate(clients.pmx.set_off)
+                    self.run_and_validate(
+                        clients.pcu.send_command,
+                        kwargs={'command': 'stop'}, timeout=None
+                    )
+                    time.sleep(5)
+                    return
```

Comment:
This block of code seems largely similar to the block of code handling `ControlState.WaitForBrake` in lines 1149-1165, with the only difference being that this block doesn't wait for the rotor to completely spin down. It might be simpler to add a boolean attribute to `ControlState.WaitForBrake` named `fast` which determines whether or not to run lines 1160-1164.

Reply:
Also whenever the `ControlState.WaitForFastBrake` finishes, it should set the state to `ControlState.Done`